subsys/net/lib/socks/socks_internal.h

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (c) 2018 Antmicro Ltd
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_SUBSYS_NET_LIB_SOCKS_SOCKS_INTERNAL_H_
+#define ZEPHYR_SUBSYS_NET_LIB_SOCKS_SOCKS_INTERNAL_H_
+
+#define SOCKS5_PKT_MAGIC			0x05
+#define SOCKS5_PKT_RSV				0x00
+
+#define SOCKS5_AUTH_METHOD_NOAUTH		0x00
+#define SOCKS5_AUTH_METHOD_GSSAPI		0x01
+#define SOCKS5_AUTH_METHOD_USERPASS		0x02
+#define SOCKS5_AUTH_METHOD_NONEG		0xFF
+
+#define SOCKS5_CMD_CONNECT			0x01
+#define SOCKS5_CMD_BIND				0x02
+#define SOCKS5_CMD_UDP_ASSOCIATE		0x03
+
+#define SOCKS5_ATYP_IPV4			0x01
+#define SOCKS5_ATYP_DOMAINNAME			0x03
+#define SOCKS5_ATYP_IPV6			0x04
+
+#define SOCKS5_CMD_RESP_SUCCESS			0x00
+#define SOCKS5_CMD_RESP_FAILURE			0x01
+#define SOCKS5_CMD_RESP_NOT_ALLOWED		0x02
+#define SOCKS5_CMD_RESP_NET_UNREACHABLE		0x03
+#define SOCKS5_CMD_RESP_HOST_UNREACHABLE	0x04
+#define SOCKS5_CMD_RESP_REFUSED			0x05
+#define SOCKS5_CMD_RESP_TTL_EXPIRED		0x06
+#define SOCKS5_CMD_RESP_CMD_NOT_SUPPORTED	0x07
+#define SOCKS5_CMD_RESP_ATYP_NOT_SUPPORTED	0x08
+
+struct socks5_method_request_common {
+	u8_t ver;
+	u8_t nmethods;
+};
+
+struct socks5_method_request {
+	struct socks5_method_request_common r;
+	u8_t methods[255];
+};
+
+struct socks5_method_response {
+	u8_t ver;
+	u8_t method;
+};
+
+struct socks5_ipv4_addr {
```

Comment:
The socks5_ipv4_addr and socks5_ipv6_addr can be removed as "struct sockaddr" has both IP address and port number information in it.

Reply:
But `struct sockaddr` also has other fields, and this particular struct is used directly as part of a buffer that is being sent/received.

Is there a way to work-around this without this extra struct?

Reply:
That also makes me think whether `__packed` should not be added to the struct definitions.

Reply:
> and this particular struct is used directly as part of a buffer that is being sent/received.

> That also makes me think whether `__packed` should not be added to the struct definitions.

It it's sent/received over network, you definitely should.

Reply:
Ok, just pushed an update with the attribute added to all the structs used as network buffers.

Reply:
> But `struct sockaddr` also has other fields, and this particular struct is used directly as part of a buffer that is being sent/received.

Hmm, if this is only used here and if you think making the struct as small as possible here is important, then we can probably do it like this (at least for now).

Reply:
> Hmm, if this is only used here and if you think making the struct as small as possible here is important, then we can probably do it like this (at least for now).

Never mind my comment, I did not realize that the struct was used to cast data from network. Then of course we need to do it like you suggested.